src/main/java/com/dell/pravega/perf/WriterWorker.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package com.dell.pravega.perf;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * abstract class for Writers.
+ */
+public abstract class WriterWorker extends Worker implements Callable<Void> {
+    final private performance perf;
+    final private EventsController eCnt;
+
+    WriterWorker(int sensorId, int events, int secondsToRun,
+                 boolean isRandomKey, int messageSize, long start,
+                 PerfStats stats, String streamName, int eventsPerSec, boolean wNr) {
+
+        super(sensorId, events, secondsToRun,
+                messageSize, start, stats,
+                streamName, 0);
+        this.eCnt = new EventsController(start, eventsPerSec);
+        perf = secondsToRun > 0 ? (wNr ? new EventsWriterTimeRW() : new EventsWriterTime()) :
+                (wNr ? new EventsWriterRW() : new EventsWriter());
+    }
+
+    /**
+     * writes the data and benchmark
+     *
+     * @param data   data to write
+     * @param record to call for benchmarking
+     * @return time return the data sent time
+     */
+    public abstract long recordWrite(String data, TriConsumer record);
+
+    /**
+     * writes the data and benchmark
+     *
+     * @param data data to write
+     */
+    public abstract void writeData(String data);
+
+
+    /**
+     * flush the producer data.
+     */
+    public abstract void flush();
+
+    @Override
+    public Void call() throws InterruptedException, ExecutionException, IOException {
+        perf.benchmark();
+        return null;
+    }
+
+    private class EventsWriter implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException, IOException {
+            // Construct event payload
+            String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
+            String payload = String.format("%-" + messageSize + "s", val);
+
+            for (int i = 0; i < events; i++) {
+                recordWrite(payload, stats::recordTime);
+                eCnt.control(i);
+            }
+            flush();
+        }
+    }
+
+    private class EventsWriterRW implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException, IOException {
+            for (int i = 0; i < events; i++) {
+                // Construct event payload
+                String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
```

Comment:
In EventsWriter above this is above the loop here it is inside. This is quite different behavior. Is it intended? If so it may be better to create a random object and call `rand.nextInt(200)` rather than multiplying a float each time.

Reply:
yes , its done